velox/expression/tests/SignatureBinderTest.cpp

```diff
@@ -150,18 +150,52 @@ TEST(SignatureBinderTest, decimals) {
       ASSERT_EQ(e.message(), "Missing constraint for variable r_precision");
     }
   }
-  // Error: Do not use short_decimal or long_decimal
+
+  // Scalar function signature, same precision and scale.
   {
-    try {
-      auto signature = exec::FunctionSignatureBuilder()
-                           .returnType("decimal(r_precision, r_scale)")
-                           .argumentType("short_decimal(a_precision, a_scale)")
-                           .argumentType("DECIMAL(b_precision, b_scale)")
-                           .build();
-      FAIL();
-    } catch (const VeloxUserError& e) {
-      ASSERT_EQ(e.message(), "Use 'DECIMAL' in the signature.");
-    }
+    auto signature = exec::FunctionSignatureBuilder()
+                         .returnType("boolean")
+                         .argumentType("SHORT_DECIMAL(a_precision, a_scale)")
+                         .argumentType("SHORT_DECIMAL(a_precision, a_scale)")
+                         .build();
+    const std::vector<TypePtr> argTypes{DECIMAL(11, 5), DECIMAL(11, 5)};
+    exec::SignatureBinder binder(*signature, argTypes);
+    ASSERT_TRUE(binder.tryBind());
+    ASSERT_TRUE(binder.tryResolveReturnType() != nullptr);
```

Comment:
Also show that `const std::vector<TypePtr> argTypes{DECIMAL(21, 5), DECIMAL(21, 5)};` will not bind?
Same below.